client/router.js

```diff
@@ -352,14 +381,51 @@ Router.prototype._invalidateTracker = function() {
 Router.prototype._updateCallbacks = function () {
   var self = this;
 
-  // add global middleware
   self._page.callbacks = [];
+  self._page.exits = [];
+
+  // add global enter triggers
+  self._page("*", function(ctx, next) {
+    _.each(self._triggersEnter, function(fn) {
+      if (typeof fn === 'function') {
+        fn(ctx);
```

Comment:
What if we call triggersEnter inside our main tracker rather than, doing it inside `page(*)
see: https://github.com/delgermurun/flow-router/blob/improved-middlewares/client/router.js#L313